modules/sql-engine/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -532,3 +532,96 @@ SqlDrop SqlDropZone(Span s, boolean replace) :
         return new IgniteSqlDropZone(s.end(this), ifExists, zoneId);
     }
 }
+
+SqlNode SqlAlterZone() :
+{
+    final Span s;
+    final SqlIdentifier zoneId;
+    final SqlIdentifier newZoneId;
+    SqlNodeList optionList = null;
+}
+{
+    <ALTER> { s = span(); }
+    <ZONE> zoneId = CompoundIdentifier()
+    (
+      <RENAME> <TO> newZoneId = CompoundIdentifier() {
+        return new IgniteSqlAlterZoneRenameTo(s.end(this), zoneId, newZoneId);
+      }
+      |
+      <WITH> { s.add(this); } optionList = AlterZoneOptions() {
+        return new IgniteSqlAlterZoneWith(s.end(this), zoneId, optionList);
+      }
+    )
+}
+
+SqlNodeList AlterZoneOptions() :
+{
+  List<SqlNode> list = new ArrayList<SqlNode>();
+  final Span s = Span.of();
+}
+{
+  AlterZoneOption(list)
+  (
+      <COMMA> { s.add(this); } AlterZoneOption(list)
+  )+
```

Comment:
did you mean `*`? With current rule it's impossible to change exactly one param. Also, we need to cover this case in tests too

Reply:
fixed